druntime/src/core/sys/posix/dirent.d

```diff
@@ -396,15 +396,7 @@ else version (CRuntime_Musl)
     {
     }
 
-    static if ( __USE_FILE_OFFSET64 )
-    {
-        dirent* readdir64(DIR*);
-        alias   readdir64 readdir;
-    }
-    else
-    {
-        dirent* readdir(DIR*);
-    }
+    dirent* readdir(DIR*);
```

Comment:
These cleanups for dead code, whilst understandable given the config module, are just unnecessary noise.

Reply:
I need to set `__USE_FILE_OFFSET64` to true again. If I don't remove the dead code before that it tries to use `*64` symbols which don't exist for musl.

Reply:
What about leaving `__USE_FILE_OFFSET64` to false?

It's fine that the `*64` symbols don't exist in musl, it's likely that these bindings are just cargo cult from the Glibc/UClibc code paths.  The clean-ups are warranted in their own right.

Reply:
No, it's not fine. musl only provides symbols like `readdir`, not `readdit64`. Using the latter leads to linker errors and is why https://github.com/dlang/dmd/pull/16361 was merged, as it fixed those symbol declarations on 64-bit arches.

Reply:
#21639

Reply:
Yes, which I've noted here https://github.com/dlang/dmd/pull/21249#discussion_r2249931632

That module is clearly incorrectly written for musl.  I suggest making a new code path for Musl that reflects its default 64 bit type sizes.

Reply:
I'll raise a PR to split up core.sys.posix.sys.types in the meantime.

If someone can correct the types for Musl as a follow-up to it and test, that would be appreciated.